src/Message/HpsGatewayResponseValidation.php

```diff
@@ -50,51 +45,56 @@ public static function getException($responseCode, $responseText, $response)
         case '0':
             break;
         case '-2':
-            $e = new InvalidResponseException(
+            $e = self::formResponseException(
```

Comment:
Let's change these calls to `static::formResponseException` to allow child classes to override `formResponseException` throw late static binding